ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -2951,9 +2951,11 @@ private void setStatsPropAndAlterPartitions(boolean resetStatistics, Table tbl,
       validWriteIdList = tableSnapshot.getValidWriteIdList();
       writeId = tableSnapshot.getWriteId();
     }
-    getSynchronizedMSC().alter_partitions(tbl.getCatName(), tbl.getDbName(), tbl.getTableName(),
-            partitions.stream().map(Partition::getTPartition).collect(Collectors.toList()),
-            ec, validWriteIdList, writeId);
+    if (!conf.getBoolVar(ConfVars.HIVESTATSAUTOGATHER)){
```

Comment:
Should this be in the starting of the method itself?

Reply:
makes sense, fixed.